app/frontend/wallet/shelley/shelley-bitbox02-crypto-provider.ts

```diff
@@ -0,0 +1,350 @@
+      prevOutIndex: input.outputIndex,
+    }
+  }
+
+  function prepareOutput(output: TxOutput): CardanoOutput {
+    if (output.tokenBundle.length > 0 && !isFeatureSupported(CryptoProviderFeature.MULTI_ASSET)) {
+      throw new InternalError(InternalErrorReason.BitBox02MultiAssetNotSupported, {
+        message:
+          'Sending tokens is currently not supported on BitBox02. Please wait for the firmware update or use Ledger instead.',
+      })
+    }
+
+    return {
+      encodedAddress: output.address,
+      value: output.coins.toString(),
+      scriptConfig: output.isChange
+        ? {
+          pkhSkh: {
+            keypathPayment: output.spendingPath,
+            keypathStake: output.stakingPath,
+          },
+        }
+        : undefined,
+    }
+  }
+
+  function prepareCertificate(
+    certificate: TxCertificate,
+    addressToAbsPathMapper: AddressToPathMapper
+  ): CardanoCertificate {
+    switch (certificate.type) {
+      case CertificateType.STAKING_KEY_REGISTRATION:
+        return {
+          stakeRegistration: {
+            keypath: addressToAbsPathMapper(certificate.stakingAddress),
+          },
+        }
+      case CertificateType.STAKING_KEY_DEREGISTRATION:
+        return {
+          stakeDeregistration: {
+            keypath: addressToAbsPathMapper(certificate.stakingAddress),
+          },
+        }
+      case CertificateType.DELEGATION:
+        return {
+          stakeDelegation: {
+            keypath: addressToAbsPathMapper(certificate.stakingAddress),
+            poolKeyhash: Buffer.from(certificate.poolHash, 'hex'),
+          },
+        }
+      case CertificateType.STAKEPOOL_REGISTRATION:
+        throw new UnexpectedError(UnexpectedErrorReason.UnsupportedOperationError, {
+          message: 'Stakepool registration not supported',
+        })
+      default:
+        throw new UnexpectedError(UnexpectedErrorReason.InvalidCertificateType)
+    }
+  }
+
+  function prepareShelleyWitness(witness: CardanoShelleyWitness): TxShelleyWitness {
+    return {
+      publicKey: Buffer.from(witness.publicKey),
+      signature: Buffer.from(witness.signature),
+    }
+  }
+
+  async function signTx(
+    txAux: TxAux,
+    addressToAbsPathMapper: AddressToPathMapper
+  ): Promise<TxSigned> {
+    const inputs = txAux.inputs.map((input) => prepareInput(input, addressToAbsPathMapper))
+    const outputs = txAux.outputs.map(prepareOutput)
+    const withdrawals = txAux.withdrawals.map((withdrawal) => ({
+      keypath: addressToAbsPathMapper(withdrawal.stakingAddress),
+      value: withdrawal.rewards.toString(),
+    }))
+    const certificates = txAux.certificates.map((certificate) =>
+      prepareCertificate(certificate, addressToAbsPathMapper)
+    )
+    const validityIntervalStart = txAux.validityIntervalStart
+      ? `${txAux.validityIntervalStart}`
+      : null
+
+    if (
+      txAux.auxiliaryData?.type === 'CATALYST_VOTING' &&
+      isFeatureSupported(CryptoProviderFeature.VOTING)
```

Comment:
Isn't this supposed to be `!isFeatureSupported(CryptoProviderFeature.VOTING)` instead?